week3/hackyourweather/src/components/forecast/Forecast.jsx

```diff
@@ -0,0 +1,80 @@
+import './Forecast.css';
+import React, { useState, useEffect } from 'react';
+import { useParams, Link } from 'react-router-dom';
+import {
+  AreaChart,
+  Area,
+  YAxis,
+  XAxis,
+  CartesianGrid,
+  Tooltip,
+} from 'recharts';
+
+const Forecast = () => {
+  const { cityId } = useParams();
+  const [forecast, setForecast] = useState([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState({ error: false });
+
+  const handleError = (message) => setIsError({ error: true, message });
+
+  useEffect(() => {
+    const url = `https://api.openweathermap.org/data/2.5/forecast?id=${cityId}&units=metric&APPID=${process.env.REACT_APP_OPENWEATHERMAP_API_KEY}`;
+
+    (async () => {
+      try {
+        setIsLoading(true);
+
+        const resp = await fetch(url);
+        if (!resp.ok) throw Error;
+        const data = await resp.json();
+
+        setForecast(data);
+
+        setIsError({ error: false });
+      } catch {
+        handleError(`Oops something went wrong!`);
+      } finally {
+        setIsLoading(false);
+      }
+    })();
+  }, [cityId]);
+
+  return (
+    <div className="forecast_container">
+      {isLoading && <p>Loading...</p>}
+      {isError && <p>{isError.message}</p>}
```

Comment:
If you keep it as is, it will always be true.

why  nothing is on the screen? because there is no message, it's just an empty string. 👉🏻 `""`

```suggestion
      {isError.error && <p>{isError.message}</p>}
```